src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -6632,4 +6628,41 @@ public void testConnectTimeout() {
       }));
     await();
   }
+
+  @Test
+  public void testResponseEndFutureCompletes_WithoutBody() throws Exception {
+      testResponseEndFutureCompletes(HttpServerResponse::end);
```

Comment:
@vietj this is the one failing without the code change but i added the other cases as well to cover it explicitly

Reply:
ok